src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -0,0 +1,93 @@
+package tech.cryptonomic.conseil.tezos
+
+import com.typesafe.config.Config
+import slick.ast.FieldSymbol
+import slick.jdbc.PostgresProfile
+import slick.jdbc.PostgresProfile.api._
+import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes.DataType.DataType
+import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes._
+import tech.cryptonomic.conseil.tezos.{TezosDatabaseOperations => TezosDb}
+import tech.cryptonomic.conseil.util.DatabaseUtil
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object PlatformDiscoveryOperations {
+
+  lazy val dbHandle: PostgresProfile.backend.Database = DatabaseUtil.db
+
+  private val tables = List(Tables.Blocks, Tables.Accounts, Tables.OperationGroups, Tables.Operations, Tables.Fees)
+  private val tablesMap = tables.map(table => table.baseTableRow.tableName -> table)
+
+  def getNetworks(config: Config): List[Network] = {
+    config.getObject("platforms").asScala.flatMap {
+      case (platform, strippedConf) =>
+        strippedConf.atKey(platform).getObject(platform).asScala.map {
+          case (network, _) =>
+            Network(network, network.capitalize, platform, network)
+        }.toList
+    }.toList
+  }
+
+  def getEntities(network: String)(implicit ec: ExecutionContext): Future[List[Entity]] = {
+    ApiOperations.countAll.map { counts =>
+      tablesMap.map(_._1).flatMap { tableName =>
+        counts.get(tableName).map { tableCount =>
+          Entity(
+            name = tableName,
+            displayName = makeDisplayName(tableName),
+            count = tableCount,
+            network = network
+          )
+        }.toList
+      }
+    }
+  }
+
+  private def makeDisplayName(name: String): String = {
+    name.capitalize.replace("_", " ")
+  }
+
+  def tableAttributes(tableName: String)(implicit ec: ExecutionContext): Future[List[Attributes]] = {
+    val res: DBIO[List[Attributes]] =
+      DBIO.sequence(
+        tablesMap.collect {
+          case (name, table) if name == tableName =>
+            val overallCount = TezosDb.countRows(table)
+            table.baseTableRow.create_*.map { col =>
+              for {
+                overallCnt <- overallCount
+                distinctCnt <- TezosDb.countDistinct(table.baseTableRow.tableName, col.name)
+              } yield makeAttributes(col, distinctCnt, overallCnt, tableName)
+            }
+        }.flatMap(_.toList)
+      )
+    dbHandle.run(res)
+  }
+
+  private def makeAttributes(col: FieldSymbol, distinctCount: Int, overallCount: Int, tableName: String): Attributes =
+    Attributes(
+      name = col.name,
+      displayName = makeDisplayName(col.name),
+      dataType = mapType(col.tpe.toString),
+      cardinality = distinctCount,
+      keyType = if (distinctCount == overallCount) KeyType.UniqueKey else KeyType.NonKey,
+      entity = tableName
+    )
+
+
+  private def mapType(tpe: String): DataType = {
```

Comment:
Since this is DataType's constructor function, maybe it makes more sense to place this in PlatformDiscoveryTypes.scala? Thoughts?

Reply:
I moved it.
I'm not proud of this function, I need to figure out how to get better type and map it to my enum, as matching on strings is not really good design. Probably some macro or shapeless would help, but I'm not sure yet.

Reply:
So from what I've investigated this solution is not perfect but I couldn't come up with anything better at this time. The closes I've got was something like this:
```
  def mapType(tpe: Type): DataType = {
    val clazz = tpe.classTag.runtimeClass

    val Ts = classOf[Timestamp]
    val Str = classOf[String]
    val Integer = classOf[Int]
    val Lng = classOf[Long]
    val Bool = classOf[Boolean]
    val Opt = classOf[Option[_]]
    val Flt = classOf[Float]
    val Dble = classOf[Double]
    val Bdecimal = classOf[BigDecimal]

    clazz match {
      case Ts => DataType.DateTime
      case Str => DataType.String
      case Integer => DataType.Int
      case Lng => DataType.LargeInt
      case Flt | Dble | Bdecimal => DataType.Decimal
      case Bool => DataType.Boolean
      case Opt => ???  // <- here is the problem
      case _ => DataType.String
    }
  }
```
But I cannot match options with it because this `classOf(Option[String]) == classOf(Option[Int])` is `true`. Also I cannot extract type from `Option[T]`. So possibly I could do the second round of matching but now on strings, but it would repeat what I already have.

Reply:
https://docs.scala-lang.org/overviews/reflection/typetags-manifests.html
ClassTag has the same problem, but maybe TypeTag doesn't?

Reply:
Unfortunately I couldn't make it work, there is possibility of doing it slightly better with `slick.meta.MTable`, but it would require few additional calls to the DB, which I'm not ok with - it has already a lot of queries per request, which were unavoidable. Depending on how the rest of the tickets will change current concept I may consider caching some values.